qfdmo/management/commands/import_product_from_qfdmod.py

```diff
@@ -0,0 +1,207 @@
+# A partir du fichier : https://docs.google.com/spreadsheets/d/1WIKCP3mye_ZYWVAyKecwHN5xrwaU0EUp/edit?usp=sharing&ouid=114461779294398717478&rtpof=true&sd=true
+
+# si il existe déjà un objet avec le nom, on met à jour l'identifiant_qfdmod
+# on crée les synomimes si ils n'existent pas
+
+# 1 match des sous-categories de la colonne P
+# si pas de colonne P, on crée catégorie colonne B et sous-catégorie de la colonne M
+
+
+import csv
+from functools import reduce
+from operator import or_
+from typing import List
+
+import requests
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+
+from qfdmo.models.categorie_objet import Objet
+
+produits_qfdmod_url = (
+    "https://data.ademe.fr/data-fair/api/v1/datasets/que-faire-de-mes-dechets-produits"
+    "/lines?size=10000&page=1&format=csv"
+)
+
+
+def _get_qfdmod_products():
+    qfdmod_products = []
+
+    # Télécharger le fichier CSV depuis l'URL
+    response = requests.get(produits_qfdmod_url)
+    response.raise_for_status()  # Vérifie si la requête a réussi
+
+    # Lire le fichier CSV depuis le contenu téléchargé
+    content = response.content.decode("utf-8").splitlines()
+    reader = csv.DictReader(content)
+
+    for row in reader:
+        qfdmod_products.append(row)
+
+    return qfdmod_products
+
+
+def _get_product_names_from_qfdmod_product(qfdmod_product: dict) -> List[str]:
+    product_names = [
+        word.lower().strip()
+        for word in qfdmod_product["Synonymes_existants"].split("/")
+        if word.lower().strip()
+    ]
+    product_names.append(qfdmod_product["Nom"].lower().strip())
+    product_names = list(set(product_names))
+    return product_names
+
+
+def _choose_sous_categorie(sous_categories, product_name):
+    choosen_sous_categorie = None
+    while choosen_sous_categorie is None:
+        input_text = f"\nPour le produit : {product_name}"
+        input_text += "\nChoisissez la categorie adéquat ?\n\n"
+        input_text += "\n".join(
+            [
+                f"{i + 1} : {prop.libelle} ({prop.categorie.libelle})"
+                for i, prop in enumerate(sous_categories)
+            ]
+        )
+        input_text += "\n0 : Ignorer\n\n"
+        input_text += "\n\nVotre choix : "
+        choosen_sous_categorie = input(input_text)
+        if not choosen_sous_categorie.isnumeric() or int(
+            choosen_sous_categorie
+        ) not in range(0, len(sous_categories) + 1):
+            choosen_sous_categorie = None
```

Comment:
Bon c'est qu'une commande django qui a une durée de vie limitée, donc rien d'indispensable (comme les commentaires plus haut), mais dans le cas présent je serais d'avis de lever une erreur (cf [_easier to ask forgiveness than permission_](https://realpython.com/python-lbyl-vs-eafp/#:~:text=EAFP%2C%20or%20easier%20to%20ask,exception%20and%20handle%20it%20appropriately.) 

```suggestion
        try:
          choosen_sous_categorie = input(input_text)
          if int(
            choosen_sous_categorie
        ) not in range(0, len(sous_categories) + 1):
            raise ValueError("la catégorie choisie n'existe pas")
        except ValueError:
          return None # pas sûr de ce qu'on veut retourner ici
```